### DISCUSSION TITLE
Can Ice work without screen recording permissions?

### DISCUSSION THREAD

**Discussion by nunuqq1** [jordanbaird/Ice#362] [General]
### Check Existing Issues

- [X] I have checked existing issues, and this issue is not a duplicate

### Description

I appreciate the work put into the ICE project and its open-source nature. However, I have some concerns regarding the requirement for screen recording permissions:

Privacy Concerns: Although ICE is open-source, the need for screen recording permissions raises privacy and security concerns. If this project were to be accessed by malicious developers, the granted permissions could potentially be exploited, leading to privacy violations.

Purpose of Screen Recording: Could you please clarify why screen recording permissions are necessary for ICE? Understanding the rationale behind this requirement would help address concerns.

Functionality Without Permissions: If ICE were to run without screen recording permissions, which functionalities would be impacted? Is it possible to design ICE to operate effectively without needing to obtain these permissions?

Thank you for considering this request. I look forward to your response.

**Comment by jordanbaird** [comment]
There is already another issue (#75) for this, which is why I converted this to a discussion.

To address your first point:

Even if we disabled screen recording permissions, there would be nothing stopping malicious developers from creating their own fork of the app that still required permissions and passing it off as the real thing, just due to the nature of open source. I should note that because Ice is licensed under GPL 3.0, I'm the only one who can sell it. If you see it being sold through any channels not linked to me, please let me know.

To address your second point:

Ice needs screen recording permissions for a couple reasons:

1. To get the individual images and titles of items to display in the Ice Bar and in the Menu Bar Layout settings.
2. To capture the desktop background image to overlay atop the menu bar when using rounded and split shapes in the Menu Bar Appearance settings.

To address your third point:

Ice _could_ run without these permissions, but would be pretty limited. No Ice Bar or arranging items in settings, and the most that it could do for appearance is a border, tint, and shadow.

I plan on changing the way permissions are presented to allow Ice to run in a limited state if the user doesn't want to grant them. Follow the linked issue for updates on that.

**Comment by BenjaminVanRyseghem** [comment]
I moved away from Bartender because of this permission, as providing a close source program (which was recently sold in a shady way) is a no-go.

Based on the fact that I setup the "Hidden"/"Always hidden" sections only now and then, and that I don't mind not tweaking the menu bar, I would also find it valuable to be able to run this app without that permission :smile:

Thanks for your work!
